components/signoffs/signoffController.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const signoffDao = require('./signoffDao');
+const JSONStream = require('JSONStream');
+const {
+  handleErrors,
+  ValidationError,
+  ConflictingDataError,
+} = require('../../utils/controllerUtils');
+const { signoff } = require('../../utils/urlSlugs');
+const logger = require('../../utils/logger').logger;
+
+const getSignoff = async (req, res, next) => {
+  try {
+    const signoffs = await signoffDao.getByCaseIdentifier(
+      req.params.caseIdentifier
+    );
+    if (signoffs) {
+      res.status(200).json(signoffs);
+    } else {
+      res.status(404).end();
+    }
+  } catch (e) {
+    handleErrors(e, 'Error getting signoff(s) for case', logger, next);
+  }
+};
+
+const addSignoff = async (req, res, next) => {
+  try {
+    const existingSignoff = null;
+    //const existingSignoff = await signoffDao.getByCaseIdentifier(
+    //  req.body.caseIdentifier
+    //);
+    //const validationResults = validateObjectsFromUser(req.body);
+    if (existingSignoff == null) {
+      await upsert(req.body);
+      res.status(201).end();
+    } else if (
+      existingSignoff.username == req.body.username &&
+      existingSignoff.signoff_step_name == req.body.signoff_step_name &&
```

Comment:
Are the `req.body` attributes expected to be snake_case? The rest of the app expects camelCase, which is why there's a [pg-promise setting](https://github.com/oicr-gsi/nabu/blob/b8b63537b54b573ea0a00671e12afcbf45c04af1/utils/dbUtils.js#L11) applied to all db connections which returns data in camelCase even if the db columns are snake case.